### PR TITLE
-tg- disease1

### PR DESCRIPTION
**Pull request by tigercat2000**
Wee! Readds Disease1, kills off disease2, and finally removes poop.dmi
(fucking piece of shit)

Needs WAY more testing, most of which I don't know how to do yet, because
I've never used this system. Fun!

Changes:
 - Virus2 has been removed completely.
  - Some symptoms have not been readded. Aka, none of them have been
    readded, I just wanted to get this shit ported.
  - Virology has been updated slightly to mirror -tg-'s boxstation, as the
    virus2 machines are gone.
 - Welcome back, disease1
  - The centrifuge, vials, splicer, analyzer, growerthing, have all been
    replaced by the PanD.E.M.I.C. 2220
  - No symptoms have been completely removed; However, vision symptoms
    have been disabled pending byond 510, and genetics symptoms have been
    disabled until "when I figure out how to fix them".
  - Advanced Mutation Toxin now uses a disease, which turns people from
    species, to slime person, to slime. Slime people are instantly turned
    to a slime.
  - Diseased touch now give people appendicitis. Yes, they VANT YOUR...
    appendix.
  - Virology now has a fridge with the following in it:
    - Antiviral Syringes
    - 1 Rhinovirus culture
    - 1 Flu culture
    - 1 Bottle of Mutagen
    - 1 Bottle of Plasma
    - 1 Bottle of Synaptizine
  - A few more things, which I can't remember and can't look at while writing
    this commit. Sometimes, git hurts.

TODO:
 - Fix genetics symptoms.
 - Give vampires a less shitty diseased touch.
 - Add appendicitis event.

:cl:
rscdel: Virus2 has been removed.
rscadd: Virus1 is back. Viva la revolution.
/:cl:

### CODE REVIEW

**Comment by tigercat2000** [comment]
# Do not merge pending figuring out how to coordinate with #4018

**Comment by KasparoVy** [comment]
Wicked wicked port

Does this mean we drop our meme diseases like Kingston's and Canis though?

**Comment by tigercat2000** [comment]
@KasparoVy no diseases have been ported to -tg-'s disease system, someone else can decide what to pick and grab, it's very easy to add new stuff

**Comment by KasparoVy** [comment]
@tigercat2000 Awesome, thanks for answering so fast! I was just curious. Really digging the modularity of it all

**Comment by Fox-McCloud** [comment]
I don't personally like the removal of the poo procs, but I'm not going to give you shit about it :stuck_out_tongue: 

Unless of course, it has enough support that people want the proc at least included for admin shennanigans or what have you =p

Definitely a :+1: to this PR--it's a better maintained system that's easier to expand on---and puts us one step closer to abductor =p

File: code/datums/diseases/advance/advance.dm [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+*/
+
+var/list/archive_diseases = list()
+
+// The order goes from easy to cure to hard to cure.
+var/list/advance_cures = 	list(
+									"sodiumchloride", "sugar", "orangejuice",
+									"spaceacillin", "salglu_solution", "ethanol",
+									"leporazine", "synaptizine", "lipolicide",
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/advance.dm]: we don't have leporazine. We do have teporone, though.

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	level = 6
+	severity = 0
+
+/datum/symptom/sensory_restoration/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB * 3))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(2)
+				if(M.reagents.get_reagent_amount("inacusiate")<10)
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/sensory.dm]: not a reagent; this is handled by oculine now.

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+				M << "<span class='warning'><b>You can't feel anything.</b></span>"
+				if(prob(10))
+					M.reagents.add_reagent("morphine",rand(5,7))
+			if(3)
+				M.reagents.add_reagent("ethanol",rand(5,7))
+				M << "<span class='warning'><b>You feel absolutely hammered.</b></span>"
+				if(prob(15))
+					M.reagents.add_reagent("morphine",rand(5,7))
+			if(4)
+				M.reagents.add_reagent_list(list("ethanol",rand(7,15),"mindbreaker",rand(5,10)))
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/sensory.dm]: LSD, not mindbreaker

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+				M << "<span class='warning'><b>You try to focus on not dying.</b></span>"
+				if(prob(20))
+					M.reagents.add_reagent("morphine",rand(5,7))
+			if(5)
+				M.reagents.add_reagent_list(list("impedrezene",rand(5,15),"ethanol",rand(7,20),"mindbreaker",rand(5,15)))
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/sensory.dm]: haloperidol and LSD not imped and mindbreaker

File: code/datums/diseases/advance/symptoms/vomit.dm [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+*/
+
+/datum/symptom/vomit/projectile
+
+	name = "Projectile Vomiting"
+	stealth = -2
+	level = 4
+
+/datum/symptom/vomit/projectile/Vomit(mob/living/carbon/M)
+	M.vomit(6,0,1,5,1)
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/vomit.dm]: OH GOSH

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 6]
```diff
@@ -198,20 +197,20 @@ var/global/list/image/splatter_cache=list()
...
-                                var/obj/effect/decal/cleanable/blood/b = new /obj/effect/decal/cleanable/blood/splatter(src.loc)
-                                b.basecolor = src.basecolor
-                                b.update_icon()
-                                for(var/datum/disease2/disease/D in src.viruses)
-                                        var/datum/disease2/disease/ND = D.getcopy()
-                                        b.viruses += ND
-
-                        if (step_to(src, get_step(src, direction), 0))
-                                break
+	set waitfor = 0
```
Comment (Fox-McCloud) [inline code/game/objects/effects/decals/Cleanable/humans.dm]: we're going to use this now? *shrug*

Comment (tigercat2000) [inline code/game/objects/effects/decals/Cleanable/humans.dm]: it's copypasted from -tg- due to lazyness, i can replace it with a spawn, but it probably doesn't matter given that I don't think anything calls this rapidly

**Comment by Dinarzad** [comment]
Anything is better then the Bay-style Virology.
I know this'll be a while coming, due to porting it over and making it play nice, but god it cannot happen soon enough IMO.
So much yes.

**Comment by Aurorablade** [comment]
BRING BACK POO BRING BACK POO BRING BACK POO

**Comment by Crazylemon64** [comment]
DING DONG THE POO IS DEAD :tada: :bell:

**Comment by tigercat2000** [comment]
@Fox-McCloud Not going to make significantly balance altering yadayada changes, I'll leave it to you for a roboburger PR and readd xenomicrobes. I did test the viruses, though, and they work. Also changed so that slime cores do make you a slime person, then into a slime.

**Comment by Regen1** [comment]
YEEEEEEEEEEEEEEEEEEEEEEEEEEEEESSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

FUCKING FINALLY

**Comment by NTSAM** [comment]
too bad we couldn't throw our poo or it would be worth keeping

File: code/datums/diseases/advance/advance.dm (hunk 1 above)
Comment (tigercat2000) [inline code/datums/diseases/advance/advance.dm]: That's a problem. A shitload of things rely on synaptizine in here.

File: code/datums/diseases/advance/symptoms/sensory.dm (hunk 2 above)
Comment (tigercat2000) [inline code/datums/diseases/advance/symptoms/sensory.dm]: @Fox-McCloud it's intended to heal ear/hearing damage, does oculine do that?

File: code/controllers/Processes/diseases.dm [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	disease_master = src
+
+	register_diseases() //register all pre-round diseases created
+
+/datum/controller/process/diseases/statProcess()
+	..()
+	stat(null, "[processing_diseases.len] diseases")
+
+
+/datum/controller/process/diseases/doWork()
```
Comment (Fox-McCloud) [inline code/controllers/Processes/diseases.dm]: This is really weird, compared with other controllers---but I realize this is exactly how TG does it.

File: code/datums/diseases/_MobProcs.dm [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+		return*/
+
+	var/target_zone = pick(head_ch;1,body_ch;2,hands_ch;3,feet_ch;4)
+
+	if(istype(src, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = src
+
+		switch(target_zone)
+			if(1)
+				if(isobj(H.head) && !istype(H.head, /obj/item/weapon/paper))
```
Comment (Fox-McCloud) [inline code/datums/diseases/_MobProcs.dm]: they actually made a snowflake check for paper hats...

lol.

File: code/datums/diseases/beesease.dm [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+				affected_mob.visible_message("<span class='danger'>[affected_mob] buzzes.</span>", \
+												"<span class='userdanger'>Your stomach buzzes violently!</span>")
+			if(prob(5))
+				affected_mob << "<span class='danger'>You feel something moving in your throat.</span>"
+			if(prob(1))
+				affected_mob.visible_message("<span class='danger'>[affected_mob] coughs up a swarm of bees!</span>", \
+													"<span class='userdanger'>You cough up a swarm of bees!</span>")
+				new /mob/living/simple_animal/hostile/poison/bees(affected_mob.loc)
+		//if(5)
+		//Plus if you die, you explode into bees
```
Comment (Fox-McCloud) [inline code/datums/diseases/beesease.dm]: Don't worry guys, I only exploded into bees.

File: code/datums/diseases/gbs.dm [hunk 10]
```diff
@@ -0,0 +1,41 @@
+/datum/disease/gbs
```
Comment (Fox-McCloud) [inline code/datums/diseases/gbs.dm]: Earth and I's favorite disease1 virus.

File: code/controllers/Processes/diseases.dm (hunk 7 above)
Comment (tigercat2000) [inline code/controllers/Processes/diseases.dm]: Not sure what you mean.

File: code/modules/mob/living/carbon/life.dm [hunk 11]
```diff
@@ -84,12 +84,7 @@
 
 	if(breath)
 		loc.assume_air(breath)
-		//spread virus2
-		if(virus2.len > 0)
-			if(prob(10) && get_infection_chance(src))
-				for(var/mob/living/carbon/M in view(1,src))
-					src.spread_disease_to(M)
-
+		air_update_turf()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/life.dm]: This is actually unrelated to disease1--not sure if we really need this or not: https://github.com/tgstation/-tg-station/pull/16161

Comment (tigercat2000) [inline code/modules/mob/living/carbon/life.dm]: Calls to assume_air() should also call air_update_turf(), iirc

File: code/datums/diseases/gbs.dm (hunk 10 above)
Comment (tigercat2000) [inline code/datums/diseases/gbs.dm]: Oh, right, forgot to remove this file. :wink:

File: code/controllers/Processes/diseases.dm (hunk 7 above)
Comment (Fox-McCloud) [inline code/controllers/Processes/diseases.dm]: Ahh, what I' saying is, that most disease processes use an istypeless `for` loops and not a `while` loop.

File: code/datums/diseases/_MobProcs.dm [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+		hands_ch = 0
+		feet_ch = 100
+
+	if(prob(15/D.permeability_mod))
+		return
+
+/*	if(satiety>0 && prob(satiety/10)) // positive satiety makes it harder to contract the disease.
+		return*/
+
+	var/target_zone = pick(head_ch;1,body_ch;2,hands_ch;3,feet_ch;4)
```
Comment (tigercat2000) [inline code/datums/diseases/_MobProcs.dm]: what in sam fucking hell is this

Comment (Fox-McCloud) [inline code/datums/diseases/_MobProcs.dm]: see above, I believe.

Comment (tigercat2000) [inline code/datums/diseases/_MobProcs.dm]:
```
pick proc
See also: 
prob proc 
Format: 
pick(Val1,Val2,...) 
Returns: 
One of the given values randomly chosen. 
To make a particular value more or less likely to be chosen, a relative probability may be specified like this: 

prob(P); Val
  Or
P; Val
A value for P of 200 makes it twice as likely as the norm, 50 half as likely, and so on. 
```
goddamnit byond

File: code/controllers/Processes/diseases.dm (hunk 7 above)
Comment (Tastyfish) [inline code/controllers/Processes/diseases.dm]: I would have to test it to be sure, but I'm pretty sure the weird stack method this uses is a lot less efficient than a for loop, and then completely cutting the entire list.

File: code/datums/diseases/_MobProcs.dm (hunk 12 above)
Comment (Tastyfish) [inline code/datums/diseases/_MobProcs.dm]: It's entertaining to read tiger criticizing his own PR.

File: code/modules/reagents/reagent_containers/glass/bottle.dm [hunk 13]
```diff
@@ -299,4 +299,120 @@
...
-		reagents.add_reagent(pick("polonium","initropidril","concentrated_initro","pancuronium","sodium_thiopental","ketamine","sulfonal","amanitin","coniine","curare","sarin","histamine","venom","cyanide","spidereggs"), 40)
\ No newline at end of file
+		reagents.add_reagent(pick("polonium","initropidril","concentrated_initro","pancuronium","sodium_thiopental","ketamine","sulfonal","amanitin","coniine","curare","sarin","histamine","venom","cyanide","spidereggs"), 40)
+
+
+/obj/item/weapon/reagent_containers/glass/bottle/plasma
+	name = "liquid plasma bottle"
+	desc = "A small bottle of liquid plasma. Extremely toxic and reacts with micro-organisms inside blood."
+	icon_state = "bottle8"
+	list_reagents = list("plasma" = 30)
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: we'll have to can all the `New` procs some time in favor if this.

File: code/modules/mob/living/carbon/carbon.dm [hunk 14]
```diff
@@ -92,6 +92,42 @@
 
 #undef STOMACH_ATTACK_DELAY
 
+/mob/living/carbon/proc/vomit(var/lost_nutrition = 10, var/blood = 0, var/stun = 1, var/distance = 0, var/message = 1)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/carbon.dm]: You know which PR I'm going to add in after this gets merged, don't you? =p

File: code/game/machinery/computer/buildandrepair.dm [hunk 15]
```diff
@@ -54,6 +54,10 @@
 /obj/item/weapon/circuitboard/med_data
 	name = "Circuit board (Medical Records)"
 	build_path = /obj/machinery/computer/med_data
+/obj/item/weapon/circuitboard/pandemic
```
Comment (Fox-McCloud) [inline code/game/machinery/computer/buildandrepair.dm]: Don't forget to add this to R&D

File: code/modules/reagents/reagent_containers/glass/bottle.dm [hunk 16]
```diff
@@ -299,4 +299,120 @@
...
+	icon_state = "bottle3"
+	spawned_disease = /datum/disease/brainrot
+
+/obj/item/weapon/reagent_containers/glass/bottle/magnitis
+	name = "Magnitis culture bottle"
+	desc = "A small bottle. Contains a small dosage of Fukkos Miracos."
+	icon_state = "bottle3"
+	spawned_disease = /datum/disease/magnitis
+
+/obj/item/weapon/reagent_containers/glass/bottle/wizarditis
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: one of these is in the MagiVend

File: code/modules/mob/living/carbon/carbon.dm (hunk 14 above)
Comment (Spacemanspark) [inline code/modules/mob/living/carbon/carbon.dm]: Don't you dare.

File: code/datums/diseases/advance/symptoms/sensory.dm [hunk 17]
```diff
@@ -0,0 +1,103 @@
+/*
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/sensory.dm]: Leave synaptizine alone in this file...

File: code/modules/reagents/newchem/disease.dm [hunk 18]
```diff
@@ -5,10 +5,189 @@
...
+	name = "Mix Virus 9"
+	id = "mixvirus9"
+	required_reagents = list("synaptizinevirusfood" = 1)
+	level_min = 1
+	level_max = 1
+
+/datum/chemical_reaction/mix_virus/rem_virus
+	name = "Devolve Virus"
+	id = "remvirus"
+	required_reagents = list("synaptizine" = 1)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/disease.dm]: diphenhydramine

File: code/modules/reagents/newchem/disease.dm [hunk 19]
```diff
@@ -5,10 +5,189 @@
...
+	name = "Mix Virus 8"
+	id = "mixvirus8"
+	required_reagents = list("plasmavirusfood" = 1)
+	level_min = 6
+	level_max = 6
+
+/datum/chemical_reaction/mix_virus/mix_virus_9
+	name = "Mix Virus 9"
+	id = "mixvirus9"
+	required_reagents = list("synaptizinevirusfood" = 1)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/disease.dm]: diphenhydramine

File: code/modules/reagents/reagent_containers/glass/bottle.dm [hunk 20]
```diff
@@ -299,4 +299,120 @@
...
+		reagents.add_reagent(pick("polonium","initropidril","concentrated_initro","pancuronium","sodium_thiopental","ketamine","sulfonal","amanitin","coniine","curare","sarin","histamine","venom","cyanide","spidereggs"), 40)
+
+
+/obj/item/weapon/reagent_containers/glass/bottle/plasma
+	name = "liquid plasma bottle"
+	desc = "A small bottle of liquid plasma. Extremely toxic and reacts with micro-organisms inside blood."
+	icon_state = "bottle8"
+	list_reagents = list("plasma" = 30)
+
+/obj/item/weapon/reagent_containers/glass/bottle/synaptizine
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: diphenhydramine

File: code/datums/diseases/advance/advance.dm [hunk 21]
```diff
@@ -0,0 +1,456 @@
...
+
+*/
+
+var/list/archive_diseases = list()
+
+// The order goes from easy to cure to hard to cure.
+var/list/advance_cures = 	list(
+									"sodiumchloride", "sugar", "orangejuice",
+									"spaceacillin", "salglu_solution", "ethanol",
+									"teporone", "synaptizine", "lipolicide",
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/advance.dm]: diphenhydramine

File: code/datums/diseases/fake_gbs.dm [hunk 22]
```diff
@@ -0,0 +1,32 @@
+/datum/disease/fake_gbs
+	name = "GBS"
+	max_stages = 5
+	spread_text = "On contact"
+	spread_flags = CONTACT_GENERAL
+	cure_text = "Synaptizine & Sulfur"
```
Comment (Fox-McCloud) [inline code/datums/diseases/fake_gbs.dm]: diphenhydramine and sulfur

File: code/modules/reagents/newchem/disease.dm [hunk 23]
```diff
@@ -5,10 +5,189 @@
...
+	id = "mutagenvirusfood"
+	description = "mutates blood"
+	color = "#A3C00F" // rgb: 163,192,15
+
+/datum/reagent/mutagen/mutagenvirusfood/sugar
+	name = "sucrose agar"
+	id = "sugarvirusfood"
+	color = "#41B0C0" // rgb: 65,176,192
+
+/datum/reagent/synaptizine/synaptizinevirusfood
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/disease.dm]: diphenhydramine

File: code/modules/reagents/newchem/disease.dm [hunk 24]
```diff
@@ -5,10 +5,189 @@
...
+	result_amount = 3
+
+/datum/chemical_reaction/virus_food_mutagen
+	name = "mutagenic agar"
+	id = "mutagenvirusfood"
+	result = "mutagenvirusfood"
+	required_reagents = list("mutagen" = 1, "virusfood" = 1)
+	result_amount = 1
+
+/datum/chemical_reaction/virus_food_synaptizine
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/disease.dm]: diphenhydramine

File: code/modules/reagents/newchem/disease.dm [hunk 25]
```diff
@@ -5,10 +5,189 @@
...
+	result_amount = 1
+
+/datum/chemical_reaction/virus_food_plasma
+	name = "virus plasma"
+	id = "plasmavirusfood"
+	result = "plasmavirusfood"
+	required_reagents = list("plasma" = 1, "virusfood" = 1)
+	result_amount = 1
+
+/datum/chemical_reaction/virus_food_plasma_synaptizine
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/disease.dm]: diphenhydramine

**Comment by Fox-McCloud** [comment]
The CMO and Virologist both need `access_mineral_storeroom` for their ID's

And the Virologist's console should notify him when uranium becomes available.

File: code/game/machinery/computer/buildandrepair.dm (hunk 15 above)
Comment (tigercat2000) [inline code/game/machinery/computer/buildandrepair.dm]: @Fox-McCloud Don't know exactly how you add new circuits to RnD.

File: code/controllers/Processes/diseases.dm (hunk 7 above)
Comment (tigercat2000) [inline code/controllers/Processes/diseases.dm]: don't really feel like rewriting this loop when it works

File: code/game/machinery/computer/buildandrepair.dm (hunk 15 above)
Comment (Fox-McCloud) [inline code/game/machinery/computer/buildandrepair.dm]: I guess I can cover this in a follow up PR---it's pretty easy though; it's just under modules->research->designs.

**Comment by Fox-McCloud** [comment]
CMO needs mineral access too.

Also you may want to talk with @Tastyfish about the viro consoles so that they'll notify the Viro when uranium or gold has arrived.

**Comment by tigercat2000** [comment]
@Fox-McCloud cmo already has mineral access https://github.com/tigercat2000/Paradise/blob/disease1/code/game/jobs/job/medical.dm#L14-L16

**Comment by Fox-McCloud** [comment]
Ahh, thanks. Didn't notice.

**Comment by Bigtimetripod** [comment]
Woo! At long last, Viro is gettung some love. :,)

**Comment by Fox-McCloud** [comment]
https://github.com/tgstation/-tg-station/blob/master/code/modules/research/designs/comp_board_designs.dm#L63

there's the pandemic board and how to add it to R&D (easy as pie, you can copy paste, more or less, though you may want to make sure the category is the same as other computer boards)

**Comment by Necaladun** [comment]
Yay better viro!

Question for the wiki - can we just lift from TG's wiki for a start to replace our own?

**Comment by tigercat2000** [comment]
@Necaladun yes, haven't changed anything other than leaving out like three viruses (genetics virus, genetics virus symptom, vision repair symptom)

**Comment by tigercat2000** [comment]
@Necaladun oh- one more virus is missing; Jungle Fever. Couldn't figure out how to make it fucking work with the species based monkies. ¯\\\_(ツ)\_/¯

**Comment by Spacemanspark** [comment]
>Jungle Fever

Oh hell.

And thank God you didn't add that genetics disease.

**Comment by Crazylemon64** [comment]
@tigercat2000 Check if the victim has a lesser form, then set their monkey DNA block's state to 1

**Comment by tigercat2000** [comment]
@Crazylemon64 gibstation best station

**Comment by tigercat2000** [comment]
# Advanced diseases confirmed to be working now. Jungle Fever at least monkeyizes people now.

@Fox-McCloud Just going to add the design, then this will be merge ready.

File: code/modules/reagents/reagent_containers/glass/bottle.dm (hunk 16 above)
Comment (tigercat2000) [inline code/modules/reagents/reagent_containers/glass/bottle.dm]: @Fox-McCloud Wizarditis is in the contraband. Wizard would have to hack the machine

**Comment by Fox-McCloud** [comment]
Most things appear to be working.

What's not:

- Viewing the viruses on medical laptops
- Combining different blood samples and having it merge symptoms 
- Curing a virus in someone, then sticking cured blood into the pandemic to make a vaccine
 - Blood curing is not meant to be a thing on TG; you HAVE to make vaccines from cured blood with the pandemic.

File: code/datums/diseases/_MobProcs.dm (hunk 8 above)
Comment (Iamgoofball) [inline code/datums/diseases/_MobProcs.dm]: hi a lot of this code is really really old

File: code/datums/diseases/gbs.dm (hunk 10 above)
Comment (Iamgoofball) [inline code/datums/diseases/gbs.dm]: gee bee esss

File: code/datums/diseases/_MobProcs.dm (hunk 12 above)
Comment (Iamgoofball) [inline code/datums/diseases/_MobProcs.dm]: OLDCODE

File: code/datums/diseases/advance/advance.dm [hunk 26]
```diff
@@ -0,0 +1,456 @@
...
+*/
+
+var/list/archive_diseases = list()
+
+// The order goes from easy to cure to hard to cure.
+var/list/advance_cures = 	list(
+									"sodiumchloride", "sugar", "orangejuice",
+									"spaceacillin", "salglu_solution", "ethanol",
+									"teporone", "diphenhydramine", "lipolicide",
+									"silver", "gold"
```
Comment (Crazylemon64) [inline code/datums/diseases/advance/advance.dm]: Given that silver and gold are readily produced in any chem dispenser, perhaps those should be changed.

File: code/datums/diseases/advance/advance.dm [hunk 27]
```diff
@@ -0,0 +1,456 @@
...
+			R.data = data.Copy()
+		if(preserve.len)
+			R.data["viruses"] = preserve
+
+/proc/AdminCreateVirus(client/user)
+
+	if(!user)
+		return
+
+	var/i = 5
```
Comment (Crazylemon64) [inline code/datums/diseases/advance/advance.dm]: Is there some way for an admin to increase this?

File: code/datums/diseases/advance/symptoms/damage_converter.dm [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+		var/mob/living/carbon/human/H = M
+
+		var/list/parts = H.get_damaged_organs(1,1) //1,1 because it needs inputs.
+
+		if(!parts.len)
+			return
+
+		M.heal_overall_damage(get_damage, get_damage)
+		M.adjustToxLoss(get_damage*parts.len)
+
```
Comment (Crazylemon64) [inline code/datums/diseases/advance/symptoms/damage_converter.dm]: This means they'll be badly hurt if they have a lot of damaged limbs, right?

File: code/datums/diseases/advance/advance.dm (hunk 26 above)
Comment (tigercat2000) [inline code/datums/diseases/advance/advance.dm]: @Crazylemon64 no
that's the point, some are easy to cure, some take a shitload of time

File: code/datums/diseases/advance/advance.dm (hunk 27 above)
Comment (tigercat2000) [inline code/datums/diseases/advance/advance.dm]: obviously not

File: code/datums/diseases/advance/symptoms/damage_converter.dm (hunk 28 above)
Comment (tigercat2000) [inline code/datums/diseases/advance/symptoms/damage_converter.dm]: yes

File: code/datums/diseases/advance/advance.dm (hunk 26 above)
Comment (Fox-McCloud) [inline code/datums/diseases/advance/advance.dm]: Yeah; a virologist has to work to get something that's difficult to cure---there's built in balance to advanced diseases----with various traits getting pushed up and down based on what it does.

You can't have a nigh incurable hidden super virus that kills you dead and is airborne all in the same thing.

Comment (Crazylemon64) [inline code/datums/diseases/advance/advance.dm]: The comment implies that the reagents are sorted by difficulty to obtain, meaning that IF that property is used, the disease will be far more easy to cure should the disease try to have its cure be a "difficult" reagent.

File: code/datums/diseases/advance/advance.dm [hunk 29]
```diff
@@ -0,0 +1,456 @@
...
+			severity = "Unknown"
+
+
+// Will generate a random cure, the less resistance the symptoms have, the harder the cure.
+/datum/disease/advance/proc/GenerateCure(list/properties = list())
+	if(properties && properties.len)
+		var/res = Clamp(properties["resistance"] - (symptoms.len / 2), 1, advance_cures.len)
+		//world << "Res = [res]"
+		cures = list(advance_cures[res])
+
```
Comment (Crazylemon64) [inline code/datums/diseases/advance/advance.dm]: Which is used right here.
